src/components/AddItemForm.tsx

```diff
@@ -1,22 +1,38 @@
-import React from 'react';
+import React, { useState } from 'react';
 
-function AddItemForm() {
+function AddItemForm(props: any) {
+  const { setDescription, description } = props;
   const domInputId = 'new-todo-input';
+  const [captureInput, setCaptureInput] = useState<string>();
+
+  const handleInputText = () => {
+    const listDescription = [...description];
+    listDescription.push(captureInput);
+    setDescription(listDescription);
+    setCaptureInput('');
+  };
+
   return (
-    <form>
+    <form onSubmit={handleInputText}>
       <h2 className="label-wrapper">
         <label htmlFor={domInputId} className="label__lg">
           Lista de tarefas PodCodar
         </label>
       </h2>
       <input
         type="text"
+        value={captureInput}
         id={domInputId}
         className="input input__lg"
         name={domInputId}
         autoComplete="off"
+        onChange={(e) => setCaptureInput(e.target.value)}
       />
-      <button type="button" className="btn btn__primary btn__lg">
+      <button
+        type="button"
```

Comment:
dentro de form, o `type` correto desse botao seria "submit"
```suggestion
        type="submit"
```